apriltag_ros/src/continuous_detector.cpp

```diff
@@ -105,12 +178,34 @@ void ContinuousDetector::imageCallback (
   // Check for an empty image and publish an empty detection message
   if (image_rect->data.empty())
   {
-    AprilTagDetectionArray tag_detection_array;
-    tag_detection_array.header = image_rect->header;
-    tag_detections_publisher_.publish(tag_detection_array);
+    publishEmptyDetection(image_rect->header);
+    return;
+  }
+
+  // Transform the target's pose from base link to this camera's optical frame
+  if (!toOptical(base_frame_id_, image_rect->header.frame_id, tag_pose_))
+  {
+    // ROS_INFO_STREAM_THROTTLE(1.0, "Valid pose not set");
+    publishEmptyDetection(image_rect->header);
     return;
   }
 
+  // Set the camera info
+  camera_model_.fromCameraInfo(camera_info);
+  // compute the pixel values for the FOV pixel buffer (ie. resizing the FOV)
+  fov_pixel_buffer_height_ = (camera_info->height/2) - ((camera_info->height * fov_size_scaler_) / 2);
+  fov_pixel_buffer_width_ = (camera_info->width/2) - ((camera_info->width * fov_size_scaler_) / 2);
+
+  // Check if not in fov or not in detection range and publish an empty detection message
+  if (!isTagInFOV(tag_pose_) || !isTagInDetectionRange(tag_pose_))
+  {
+    // ROS_INFO_STREAM_THROTTLE(1.0, "FOV or range check not passed");
+    publishEmptyDetection(image_rect->header);
+    return;
+  }
+
+  // ROS_INFO_STREAM_THROTTLE(0.1, "Frame " << image_rect->header.frame_id << " is in the FOV, detecting");
```

Comment:
```suggestion
```

Reply:
I think this one is useful to keep for debugging purposes in the future - but I can remove if you still want me to.

Reply:
Cleared up in call, removing